### PR TITLE
My Sites: move add-new-site button to bottom of sidebar

### PR DESCRIPTION
**Pull request by mtias**
Instead of in the same spot as the "switch site" button.

image

image

cc @shaunandrews

### CODE REVIEW

**Comment by shaunandrews** [comment]
Something strange happening at shorter heights:

screen shot 2016-03-23 at 11 01 51 am

**Comment by shaunandrews** [comment]
Maybe we can match the styling of the other items on mobile:

screen shot 2016-03-23 at 11 02 34 am

**Comment by mtias** [comment]
@shaunandrews thanks for the review, made a few tweaks.

**Comment by rickybanister** [comment]
This is really nice @mtias!

Let's resurrect the PR to move the dev environment bug.

**Comment by shaunandrews** [comment]
screen shot 2016-03-25 at 11 32 06 am

I think the flex box stuff is screwing up something at shorter heights, especially noticeable on mobile sizes.

**Comment by jasmussen** [comment]
Rebased. Will also be doing a little work on this branch to merge in my generic footer efforts. Things might get a little messy during the course, but it'll get where it needs to in the end.

**Comment by jasmussen** [comment]
Merged the two branches. They're stacked right now — I'm probably going to change this. But I figured I'd show the efforts for papertrail purposes:

screen shot 2016-04-08 at 10 10 33

Going to do some refactoring next.

**Comment by jasmussen** [comment]
Again for papertrail purposes, here are mockups for stacked and non stacked:

footer

alt-footer

Going to try the non-stacked version next.

**Comment by jasmussen** [comment]
Did a bunch of stuff to try and move away from the stacked version. I apologize for it being a little bit messy, but I wanted to get the ball rolling and attempt the non-stacked version before I spend too much time on it. 

#### Screenshots:

Desktop:

screen shot 2016-04-08 at 12 15 16

Smaller desktop:

screen shot 2016-04-08 at 12 15 22

Mobile:

screen shot 2016-04-08 at 12 15 52

Terms and Privacy were nixed. That makes me a bit sad — we could probably do with just one of them, but there simply isn't room without it feeling crowded. 

If we combined Terms and Privacy into a single link, could we show that on mobile and desktop breakpoints, hide it on the in-betweens?

I also renamed "Add new WordPress" to "Add new Site". Is that kosher? 

Lay ALL your thoughts on me.

**Comment by shaunandrews** [comment]
I don't mind dropping terms and privacy, but maybe they find a new home in /me?

I don't have a strong opinion on the "add WordPress" vs. "add site," but I feel like we've tried to lean towards "add WordPress" in the past — but honestly that feels so confusing from a new users perspective.

What if we made the two links equal width, with a separator between them? Right now it almost looks like it says "Add New Site    Help", rather than two different links.

**Comment by jasmussen** [comment]
> I don't mind dropping terms and privacy, but maybe they find a new home in /me?

Yep, that's an option. I kept them in as the initial ticket I meant to "fix" with my original PR included those, and a language picker. 

> What if we made the two links equal width, with a separator between them? Right now it almost looks like it says "Add New Site Help", rather than two different links.

I think that could work.

I'd like to keep exploring the no-separators for a bit more before I do that — there are so many lines already. What if the Help button has an icon, like in this mockup?

https://cloud.githubusercontent.com/assets/1204802/14380610/bbe96404-fd80-11e5-867f-c95d0e583239.png

**Comment by rickybanister** [comment]
I think the icon helps make them equal citizens. Without it they can be a little too close to each other for comfort.

**Comment by folletto** [comment]
> I'd like to keep exploring the no-separators for a bit more before I do that — there are so many lines already. What if the Help button has an icon, like in this mockup?

The overall balance seems good there.  :)

**Comment by mtias** [comment]
@jasmussen tweaked a bit the usage of sidebar-footer and changed the link to the calypso section for "help".

File: client/my-sites/sidebar/sidebar.jsx [hunk 1]
```diff
@@ -712,6 +711,9 @@ module.exports = React.createClass( {
 					</SidebarMenu>
 					: null
 				}
+				<SidebarFooter>
```
Comment (jasmussen) [inline client/my-sites/sidebar/sidebar.jsx]: I know "valid HTML" is mostly an intellectual thing at this point — but right now the rendered markup looks (simplified) like this: 

```
<ul class="sidebar">
	<div class="card current-site"> ... </div>
	<li class="sidebar__menu">
		<ul>
			<li class="stats"> ... </li>
			...
		</ul>
	</li>
	<div class="sidebar__footer"> ... </div>
</ul>
```

So there are a bunch of `div`s as direct descendants of an `ul`. I thought that wasn't kosher?

**Comment by jasmussen** [comment]
Thanks a bunch for your code cleanup Matias, appreciate it. 

I added a gridicon next to the help icon:

screen shot 2016-04-13 at 10 41 50

In the future, we might hook into this footer area and tune the help link, or add other links (like a language switcher). But for now I feel like this is visually in a good place to go! Thoughts?

**Comment by folletto** [comment]
:+1:

File: client/my-sites/sidebar/sidebar.jsx (hunk 1 above)
Comment (mtias) [inline client/my-sites/sidebar/sidebar.jsx]: We need to get rid of the main `ul` for the sidebar.

**Comment by mtias** [comment]
Sounds good to me. We can experiment with language switcher separately.

File: client/my-sites/sidebar/sidebar.jsx (hunk 1 above)
Comment (jasmussen) [inline client/my-sites/sidebar/sidebar.jsx]: > We need to get rid of the main ul for the sidebar.

Smells like a separate PR.

**Comment by jasmussen** [comment]
Found a little issue on the mobile responsive profile page that I'm going to tune a bit.

**Comment by jasmussen** [comment]
On my second deep dive test of this sidebar update, I noticed that we still hadn't taken quite care of the collapsing "Sign Out" button issue that Shaun previously brought up. Seems like the way the button is built, isn't conducive to how flex deals with magic margins. 

I pushed a fix that makes it look good just now:

screen shot 2016-04-13 at 12 15 40

I feel like we might need to look at the button CSS so we don't have to add min-heights to them to preven them from collapsing. But is that best done here, or in a separate PR?

Apologies, my folks. Once more unto the review tag.

**Comment by mtias** [comment]
> But is that best done here, or in a separate PR?

Whatever you prefer. We need to polish that area so probably a separate PR is better.

**Comment by jasmussen** [comment]
> We need to polish that area so probably a separate PR is better.

I agree. I think it'd be good to get this in.

**Comment by mtias** [comment]
:+1:

**Comment by lancewillett** [comment]
Hi @jasmussen — in case it's related, seeing issues in Safari after this change.

1. Sign Out button is too wide now, Safari 9.1 on OS X 10.11.4.

screen shot 2016-04-13 at 09 07 09

2. #4704 — two Help links next to each other.

3. Safari and Chrome breakage near top of My Sites picker.

screen shot 2016-04-13 at 08 52 37

**Comment by lancewillett** [comment]
Mobile styles look broken as well, testing with iPhone 6+ on horizon.wordpress.com

**Comment by jasmussen** [comment]
Almost have a fix.

**Comment by jasmussen** [comment]
@lancewillett Fix deployed. Let me know if you run into any issues I couldn't catch in my tests!

**Comment by lancewillett** [comment]
Looking good, testing in Horizon again. Thank for quick fixes, Joen.

screen shot 2016-04-13 at 11 39 41

screen shot 2016-04-13 at 11 40 13

**Comment by jasmussen** [comment]
Thank you for great testing, and catching the regressions quickly!

**Pull request merged**